### ISSUE TITLE
[Mailer] `mailer.envelope.recipients` configuration does not work with sendmail

### ISSUE THREAD

**Issue by alexey-muzyka-1** [symfony/symfony#46877]
### Symfony version(s) affected

5.4.9

### Description

Symfony/Mailer : The recipients feature ("Always Send to the same Address") does not work with sendmail. **PHP 8.0**

### How to reproduce

add to: config/packages/dev/mailer.yaml
```yaml
framework:
    mailer:
        dsn: '%env(MAILER_DSN)%'
        envelope:
            recipients: ['valid@example.com']
            sender: 'no-reply@test.com'
```

Create any Email sender
```php
class EmailSender
{
    public function __construct(
        private MailerInterface $mailer,
    ) {
    }

    /**
     * @throws TransportExceptionInterface
     */
    public function send(): void
    {
        $email = (new TemplatedEmail())
            ->to('invalid@test.com')
            ->subject('Test Email)
            ->htmlTemplate('email/template.html.twig');

        $this->mailer->send($email);
    }
}
```

Process email. I've used mailhog to catch emails. So response is:

![image](https://user-images.githubusercontent.com/49100406/177784346-7c29360f-0b77-4e3b-8931-d7dbfaf213d6.png)

As you could see. The **sender** was retrieved from the configurations, but the **recipient** is not

### Additional Context

It was fixed in [this issue](https://github.com/symfony/symfony/issues/41149)

**Comment by xabbuh** [comment]
Can you please try to debug why the fix done in #45690 doesn't work in your case?

**Comment by alexey-muzyka-1** [comment]
I could. And for now just an assumption.

Fix was provided, as I see, only for `SendmailTransport.php` and I use `SmtpTransport.php` since my MAILER_DSN is `smtp://mailhog:1025`

Will check it to be sure a little bit later

**Comment by xabbuh** [comment]
@alexmusyka Did you manage to find out something?

**Comment by xabbuh** [comment]
I took your example and debugged this a bit. I think there is nothing to fix here. The (E)SmtpTransport uses the recipient address from the envelope to actually send the e-mail. This address can be totally independent of the value provided with the `To` header which we IMO should not modify.

**Comment by fabpot** [comment]
Closing then, thank you for investigating @xabbuh

**Comment by DanyDes** [comment]
I've been working recently with Symfony Mailer because we used Swift Mailer. While configuring the DEV environment I was worried about not sending real emails so I added the config in the dev/mailer.yaml. After sending some emails I saw the recipients list and the DEV recipient as CCO in the email sent. I was worried because I thought all of them had recieved the email.

Swift Mailer only lists the DEV recipients but Symfony Mailer lists all recipients on the email sent. I checked all recipients and only the DEV recipients got the email.

NOTE: This is just a clarification for other developers.
